src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -457,11 +477,52 @@ export class MeetingManager implements AudioVideoObserver {
         this.selectAudioInputDeviceError = error;
         console.error('Failed to choose audio input device.', error);
       }
-      this.selectedAudioInputDevice = deviceId;
+  
+      let innerDevice = null;
+      if (isAudioTransformDevice(device)) {
+        innerDevice = await device.intrinsicDevice();
+      } else {
+        innerDevice = device;
+      }
+
+      if (is440) {
+        this.selectedAudioInputDevice = '440';
+      } else if (innerDevice === null) {
+        this.selectedAudioInputDevice = null;
+      } else if (typeof innerDevice === 'string') {
+        this.selectedAudioInputDevice = innerDevice;
+      } else if (innerDevice instanceof MediaStream) {
+        this.selectedAudioInputDevice = innerDevice.id;
+      } else {
+        const deviceId = this.getDeviceID(innerDevice);
+        this.selectedAudioInputDevice = deviceId;
+      }
+
+      this.selectedAudioInputTransformDevice = device;
     }
     this.publishSelectedAudioInputDevice();
     this.publishSelectAudioInputDeviceError();
-  };
+    this.publishSelectedAudioInputTransformDevice();
+  };  
+
+  private getDeviceID = (innerDevice: MediaTrackConstraints| MediaTrackConstraintSet): string => {
```

Comment:
Did you see https://github.com/aws/amazon-chime-sdk-js/blob/747c2b3201f2fe694eb77ce8e35d99384fcee32d/src/devicecontroller/DefaultDeviceController.ts#L968 ?

Reply:
Sorry, but I can not open this link...

Reply:
Oops! Try https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/DefaultDeviceController.ts#L972.